app/src/main/java/com/vero/tasky/auth/util/PasswordErrorParser.kt

```diff
@@ -3,26 +3,26 @@ package com.vero.tasky.auth.util
 import com.vero.tasky.R
 import com.vero.tasky.auth.domain.usecase.password.PasswordValidationResult
 
-object PasswordValidator {
+object PasswordErrorParser {
 
-    fun validatePassword(
+    fun parse(
         passwordValidationResult: PasswordValidationResult,
-    ) : ValidationResult {
+    ) : PasswordParsedResult {
         return when (passwordValidationResult) {
             PasswordValidationResult.SUCCESS -> {
-                ValidationResult(isValid = true)
+                PasswordParsedResult.Valid
             }
             PasswordValidationResult.TOO_SHORT -> {
-                ValidationResult(error = R.string.password_is_too_short)
```

Comment:
I actually liked the ValidationResult

Since it could be used for a Validation logic all across the app

If error, you return a message. Else, you return valid

Think about it
You can use it for password validation, but also...
If in the future you want to validate if something is correct (ex: A user exists) you can also use ValidationResult for success, or error message if doesnt exist

And for all the validation logic in the app